Mlem/App/Views/Pages/ReportComposerView.swift

```diff
@@ -0,0 +1,145 @@
+        Section {
+            HStack(spacing: 12) {
+                ForEach([
+                    LocalizedStringResource("Spam"),
+                    LocalizedStringResource("Troll"),
+                    LocalizedStringResource("Abuse")
+                ], id: \.key) { item in
+                    Text(item)
+                        .padding(.vertical, 8)
+                        .frame(maxWidth: .infinity)
+                        .background(palette.secondaryGroupedBackground, in: .rect(cornerRadius: 10))
+                        .contentShape(.rect)
+                        .onTapGesture {
+                            var item = item
+                            // TODO: Set this to instance/community language?
+                            item.locale = .init(languageCode: .english)
+                            reason = String(localized: item)
+                        }
+                }
+            }
+            .listRowBackground(Color.clear)
+            .listRowInsets(.init())
+        }
+    }
+    
+    @ViewBuilder
+    func ruleList(_ profilable: any Profile2Providing) -> some View {
+        let rules = [BlockNode](profilable.description ?? "").rules()
+        if rules.count >= 1 {
+            Section {
+                ForEach(Array(rules.enumerated()), id: \.offset) { index, blocks in
+                    HStack(spacing: 12) {
+                        Image(systemName: "\(index + 1).circle.fill")
+                            .foregroundStyle(palette.secondary)
+                            .fontWeight(.semibold)
+                        Markdown(blocks, configuration: .default)
+                            .frame(maxWidth: .infinity)
+                    }
+                    .contentShape(.rect)
+                    .onTapGesture {
+                        switch blocks.first {
+                        case let .paragraph(inlines: inlines), .heading(level: _, inlines: let inlines):
+                            let text = inlines.stringLiteral
+                            if text.count < 100 {
+                                reason = "\(profilable.name) rule #\(index + 1): \"\(text)\""
+                                return
+                            }
+                        default:
+                            break
+                        }
+                        reason = "\(profilable.name) rule #\(index + 1)"
+                    }
+                }
+            } header: {
+                HStack {
+                    CircleCroppedImageView(profilable)
+                        .frame(width: 22, height: 22)
+                    Text("\(profilable.name) rules:")
+                        .foregroundStyle(palette.secondary)
+                        .textCase(nil)
+                }
+            }
+        }
+    }
+    
+    func send() async {
+        do {
+            try await target.report(reason: reason)
+            HapticManager.main.play(haptic: .success, priority: .low)
```

Comment:
Build failure on Xcode 15.4

```suggestion
            await HapticManager.main.play(haptic: .success, priority: .low)
```